paasta_tools/cli/cmds/spark_run.py

```diff
@@ -617,13 +632,13 @@ def configure_and_run_docker_container(
                 file=sys.stderr,
             )
 
+    original_docker_cmd = args.cmd or instance_config.get_cmd()
     spark_ui_port = pick_random_port(args.service + str(os.getpid()))
-    spark_app_name = "paasta_spark_run_{}".format(get_username())
+    spark_app_name = get_spark_app_name(original_docker_cmd)
+
     container_name = spark_app_name + "_" + str(spark_ui_port)
-    original_docker_cmd = args.cmd or instance_config.get_cmd()
     if "jupyter" not in original_docker_cmd:
```

Comment:
Can you unify the logic so that you can also account for `jupyter` in the new function you made?

Reply:
haha, I actually found this "jupyter" is not really in use. because we rename the command to be "jupyter-lab". 

It also seems fine to append port to jupyter-lab, so I am removing this section. However, in order to identify the app from spark history server, I am also making all these type to have name starts as "paasta_jupyter"